### PR TITLE
PKG-4348 streamlit-extras 0.4.0

### PR DESCRIPTION
**Pull request by psteyer**
streamlit-extras 0.4.0 :snowflake:

**Destination channel:** defaults

### Links

- [PKG-4348](https://anaconda.atlassian.net/browse/PKG-4348) 
- [Upstream repository](https://github.com/arnaudmiribel/streamlit-extras)
- [Upstream changelog/diff](https://github.com/arnaudmiribel/streamlit-extras/compare/v0.2.7...v0.4.1)

### Explanation of changes:

- Updated SHA and Version number
- removed patch and patch dependencies
- reset build number
- added `entrypoints` dependency
- added `htbuilder`
- added `prometheus_client`
- removed python pinning
- updated dev url

[PKG-4348]: https://anaconda.atlassian.net/browse/PKG-4348?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,37 +1,31 @@
...
-    # not really needed. Rather than adding entrypoints to the runtime dependencies, it was decided to remove it from the implementation.
-    - patches/0001-remove-the-usage-of-entrypoints.patch
+  sha256: ac67645ab84accb5ae4de8ef7ca5dd5fc69965d934b9373f813770720814204d
 
 build:
   script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
-  number: 1
+  number: 0
   # s390x is missing streamlit
   skip: true  # [py<38 or s390x]
```
Comment (JamesRobertsonGames) [inline recipe/meta.yaml:18]: need to  replicate `<3.9.7,>3.9.7,<4.0` in skip to match this in their pyproject.toml

https://github.com/arnaudmiribel/streamlit-extras/blob/28c9d995c9b4a3ca04484fa4b3bb9e34855149f6/pyproject.toml#L20C1-L20C87

File: recipe/meta.yaml [hunk 2]
```diff
@@ -40,11 +34,12 @@ requirements:
     - streamlit-camera-input-live >=0.2.0
     - streamlit-vertical-slider >=1.0.2
     - streamlit-toggle-switch >=1.0.2
-    - htbuilder 0.6.1
+    - htbuilder
```
Comment (JamesRobertsonGames) [inline recipe/meta.yaml]: This has actually increased

https://github.com/arnaudmiribel/streamlit-extras/blob/28c9d995c9b4a3ca04484fa4b3bb9e34855149f6/pyproject.toml#L29C14-L29C21
```suggestion
    - htbuilder >=0.6.2
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,37 +1,31 @@
 {% set name = "streamlit-extras" %}
-{% set version = "0.2.7" %}
+{% set version = "0.4.0" %}
```
Comment (JamesRobertsonGames) [inline recipe/meta.yaml:2]: Just a note, unless there's a specific need for 0.4.0, then 0.4.2 should be built now

**Review by JamesRobertsonGames — CHANGES_REQUESTED** [review]
Couple of things to check 👍

File: recipe/meta.yaml (hunk 3 above)
Comment (psteyer) [reply recipe/meta.yaml:2]: I believe that snowflake asked for specifically `0.4.0`.

File: recipe/meta.yaml (hunk 2 above)
Comment (psteyer) [reply recipe/meta.yaml]: Added the pinning.

File: recipe/meta.yaml (hunk 1 above)
Comment (psteyer) [reply recipe/meta.yaml:18]: I added a python pinning of `!=3.9.7` for python under `host` and `run`

**Comment by cbouss** [comment]
I believe you are overriding the cbc value for python.
Using `- python {{ python }},!=3.9.7` might work better. 
That being said, ignoring this restriction wouldn't be the end of the world.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -40,11 +34,12 @@ requirements:
     - streamlit-camera-input-live >=0.2.0
     - streamlit-vertical-slider >=1.0.2
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - streamlit-vertical-slider >=2.5.5
```
see https://github.com/arnaudmiribel/streamlit-extras/blame/v0.4.0/pyproject.toml#L27

File: recipe/meta.yaml [hunk 5]
```diff
@@ -56,7 +51,7 @@ test:
 
 about:
   home: https://share.streamlit.io/
-  dev_url: https://pypi.org/project/streamlit-extras/
+  dev_url: https://github.com/arnaudmiribel/streamlit-extras
   doc_url: https://share.streamlit.io/
```
Comment (boldorider4) [inline recipe/meta.yaml]:
```suggestion
  doc_url:  https://github.com/arnaudmiribel/streamlit-extras
```

**Review by boldorider4 — APPROVED** [review]
LGTM but comments need to be addressed

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,50 +1,45 @@
...
     - st-annotated-text >=3.0.0
     - streamlit-embedcode >=0.1.2
     - streamlit-keyup >=0.1.9
     - protobuf !=3.20.2
     - streamlit-camera-input-live >=0.2.0
-    - streamlit-vertical-slider >=1.0.2
+    - streamlit-vertical-slider >=2.5.5
     - streamlit-toggle-switch >=1.0.2
-    - htbuilder 0.6.1
+    - htbuilder >=0.6.1
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
    # Temporarily diverging from upstream to fix a conflict that happens in a circular dependency.
    - htbuilder 0.6.1
```

File: recipe/meta.yaml [hunk 7]
```diff
@@ -1,63 +1,73 @@
...
+    # Temporarily diverging from upstream to fix a conflict that happens in a circular dependency.
     - htbuilder 0.6.1
     - streamlit-faker >=0.0.2
     - streamlit-card >=0.0.4
     - markdownlit >=0.0.5
     - streamlit-image-coordinates >=0.1.1,<1.0.0
+    - prometheus_client >=0.14.0
+    # upstream forgot to add validators to streamlit-extra 0.4.0, but require validators 0.22.0 for tests.
+    # Test pass with validators 0.18.2
+    - validators
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:52]: Should it be moved to the `test/requires` section instead?

**Review by skupr-anaconda — APPROVED** [review]
LGTM but `validators` for tests

File: recipe/meta.yaml (hunk 7 above)
Comment (psteyer) [reply recipe/meta.yaml:52]: `validators` seems to be sitting with all the other dependencies in upstream:
https://github.com/arnaudmiribel/streamlit-extras/blob/f715952cb721f57b2841af972e40e6f3eef22f45/pyproject.toml#L36

The only odd thing is that in `0.4.0` it's missing, but during the test process it seems to still be required.
https://github.com/arnaudmiribel/streamlit-extras/blob/v0.4.0/pyproject.toml#L36

Comment (psteyer) [reply recipe/meta.yaml:52]: I'll move it to the `test` section and see if it works!

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:52]: @psteyer please leave it into the runtime deps.

File: recipe/meta.yaml (hunk 3 above)
Comment (boldorider4) [reply recipe/meta.yaml:2]: because they requested `0.4.0` on April 15th, well after `0.4.2` was released, I would keep `0.4.0`.

**Pull request merged**